test/extended/apiserver/graceful_termination.go

```diff
@@ -61,6 +62,41 @@ var _ = g.Describe("[sig-api-machinery][Feature:APIServer][Late]", func() {
 		}
 	})
 
+	// This test extends the previous test by checking the content of the termination files for kube-apiservers.
+	// It should catch cases where the event is not persisted in the database. It should also catch
+	// cases where the KAS is immediately restarted or shut down after an ungraceful termination.
+	g.It("kubelet terminates kube-apiserver gracefully extended", func() {
+		var finalMessageBuilder strings.Builder
+		terminationRegexp := regexp.MustCompile(`Previous pod .* did not terminate gracefully`)
+
+		masters, err := oc.AsAdmin().KubeClient().CoreV1().Nodes().List(context.Background(), metav1.ListOptions{LabelSelector: "node-role.kubernetes.io/master"})
+		o.Expect(err).NotTo(o.HaveOccurred())
+		for _, master := range masters.Items {
+			g.By(fmt.Sprintf("Getting log files for kube-apiserver on master: %s", master.Name))
+			kasLogFileNames, _, err := oc.AsAdmin().Run("adm").Args("node-logs", master.Name, "--path=kube-apiserver/").Outputs()
+			o.Expect(err).NotTo(o.HaveOccurred())
+			for _, kasLogFileName := range strings.Split(kasLogFileNames, "\n") {
+				if isKASTerminationLogFile(kasLogFileName) {
```

Comment:
Nit: if you switch this condition to be:
```golang
if !isKASTerminationLogFile(kasLogFileName) {
    continue
}
g.By(fmt.Sprintf("Getting log files for kube-apiserver on master: %s", master.Name))
...
```
then the code reads easier having less indentation.

Reply:
sure, created https://github.com/openshift/origin/pull/29016, ptal